bluetooth/src/iosMain/kotlin/scanner/Scanner.kt

```diff
@@ -141,7 +142,10 @@ actual class Scanner internal constructor(
         discoveringDelegates.add(delegate)
         centralManager.delegate = delegate
         awaitPoweredOn.await()
-        centralManager.scanForPeripheralsWithServices(filter?.let { listOf(filter) }, null)
+        centralManager.scanForPeripheralsWithServices(
+            filter?.let { listOf(filter) },
+            mapOf(CBCentralManagerScanOptionAllowDuplicatesKey to true)
```

Comment:
I thought of that but it is iOS specific only and if you do not set it it will behave in a way Android can not (Except if I write that layer over it myself so Android also filters out same device after first discovery)

Reply:
@avdyushin took me some time but got it, I see Android has an overload on builder params for Android specific stuff. I added the same for iOS and a default null value. I guess in the future could be improved by having those options default to the same behavior. 

- https://developer.android.com/reference/android/bluetooth/le/ScanSettings
- https://developer.apple.com/documentation/corebluetooth/cbcentralmanager/peripheral_scanning_options